robottelo/virtwho_utils.py

```diff
@@ -203,14 +204,18 @@ def get_configure_option(option, filename):
         raise VirtWhoError(f"option {option} is not exist or not be enabled in {filename}")
 
 
-def _get_hypervisor_mapping(logs, hypervisor_type):
+def _get_hypervisor_mapping(hypervisor_type):
     """Analysing rhsm.log and get to know: what is the hypervisor_name
     for the specific guest.
     :param str logs: the output of rhsm.log.
     :param str hypervisor_type: esx, libvirt, rhevm, xen, libvirt, kubevirt
     :raises: VirtWhoError: If hypervisor_name is None.
     :return: hypervisor_name and guest_name
     """
+    _, logs = runcmd('cat /var/log/rhsm/rhsm.log')
+    if "Host-to-guest mapping being sent to" not in logs:
+        time.sleep(20)
+        _, logs = runcmd('cat /var/log/rhsm/rhsm.log')
```

Comment:
This could be a good case to use `wait_for` instead of a single set sleep time. You can find a number of examples in the robottelo codebase.

Reply:
> This could be a good case to use `wait_for` instead of a single set sleep time. You can find a number of examples in the robottelo codebase.

Thank you @JacobCallahan for your suggestion.  Have used the wait_for instead of sleep time.